toolchain/parse/state.def

```diff
@@ -1154,17 +1154,27 @@ CARBON_PARSE_STATE(ImplBeforeAs)
 
 // Handles the start of a `var` or `returned var`.
 //
-// var ...             (variant is not Returned)
+// var ...             (variant is Decl)
 //    ^
 //   1. BindingPattern
 //   2. VarAfterPattern
-//   3. VarFinishAs(Decl|For)
+//   3. VarFinishAsDecl
 //
-// var (...)
+// var (...)           (variant is Decl)
 //    ^
 //   1. ParamListAsRegular
 //   2. VarAfterPattern
-//   3. VarFinishAs(Decl|For)
+//   3. VarFinishAsDecl
+//
+// var ...             (variant is For)
+//    ^
+//   1. BindingPattern
+//   2. VarFinishAsFor
+//
+// var (...)           (variant is For)
+//    ^
+//   1. ParamListAsRegular
+//   2. VarFinishAsFor
```

Comment:
These comments don't quite match the code: this state uses the `Pattern` state rather than dispatching on the `(` versus no-`(` itself.

Reply:
Whoops, you're right. Looks like an incomplete edit from #3448.